Building the core repo with any Java version > 11 fails due to access errors

##### Issue Overview

Building the 

##### Expected Behaviour
The build should complete under Java SE 17 and 21

##### Current Behaviour
There is a failure in the org.jboss.arquillian.testenricher.cdi.CDIInjectionEnricherTestCase due to missing module access grants.

##### Steps To Reproduce
1. set your JAVA_HOME to any JDK version > 11
2. run mvn clean install

##### Additional Information

Anything relevant to help us resolving the problem. For example if you are using `mvn` you can share output for `mvn --version` as well as dependencies in use `mvn dependency:tree`. OS as well as JDK version would be helpful too.

For long outputs such as stacktraces please use HTML5 `<details>`

```
starksm@Scotts-Mac-Studio arq-main-core % $JAVA_HOME/bin/java -version
openjdk version "21.0.1" 2023-10-17 LTS
OpenJDK Runtime Environment Temurin-21.0.1+12 (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.1+12 (build 21.0.1+12-LTS, mixed mode)

[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.476 s <<< FAILURE! -- in org.jboss.arquillian.testenricher.cdi.CDIInjectionEnricherTestCase
[ERROR] org.jboss.arquillian.testenricher.cdi.CDIInjectionEnricherTestCase.shouldReleaseCreationalContext -- Time elapsed: 0.308 s <<< ERROR!
org.jboss.weld.exceptions.WeldException: WELD-001524: Unable to load proxy class for bean Implicit Bean [javax.enterprise.inject.Instance] with qualifiers [@Default] with class interface javax.enterprise.inject.Instance using classloader jdk.internal.loader.ClassLoaders$AppClassLoader@63c12fb0
        at org.jboss.weld.bean.proxy.ProxyFactory.getProxyClass(ProxyFactory.java:397)
        at org.jboss.weld.bean.builtin.AbstractFacadeBean.initializeAfterBeanDiscovery(AbstractFacadeBean.java:61)
        at org.jboss.weld.bootstrap.ConcurrentBeanDeployer$AfterBeanDiscoveryInitializerFactory.doWork(ConcurrentBeanDeployer.java:132)
        at org.jboss.weld.bootstrap.ConcurrentBeanDeployer$AfterBeanDiscoveryInitializerFactory.doWork(ConcurrentBeanDeployer.java:123)
        at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:62)
        at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:55)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.jboss.classfilewriter.ClassFile
        at org.jboss.weld.bean.proxy.ProxyFactory.newClassFile(ProxyFactory.java:518)
        at org.jboss.weld.bean.proxy.ProxyFactory.createProxyClass(ProxyFactory.java:473)
        at org.jboss.weld.bean.proxy.ProxyFactory.getProxyClass(ProxyFactory.java:389)
        ... 9 more
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @1cab0bfb [in thread "weld-worker-5"]
        at java.base/java.lang.reflect.AccessibleObject.throwInaccessibleObjectException(AccessibleObject.java:391)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:367)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:315)
        at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:203)
        at java.base/java.lang.reflect.Method.setAccessible(Method.java:197)
        at org.jboss.classfilewriter.ClassFile$1.run(ClassFile.java:238)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
        at org.jboss.classfilewriter.ClassFile.<clinit>(ClassFile.java:233)
        ... 12 more
```

--- a/testenrichers/cdi/pom.xml
+++ b/testenrichers/cdi/pom.xml
@@ -119,5 +119,18 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
+  
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <argLine>
+            --add-opens java.base/java.lang=ALL-UNNAMED
+          </argLine>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
 </project>
-

--- a/testenrichers/cdi/pom.xml
+++ b/testenrichers/cdi/pom.xml
@@ -27,6 +27,7 @@
     <version.javax-el>2.2</version.javax-el>
     <version.slf4j>2.0.12</version.slf4j>
 
+    <java11.opensargs></java11.opensargs>
   </properties>
 
   <!-- Dependencies -->
@@ -119,7 +120,7 @@
       <scope>test</scope>
     </dependency>
   </dependencies>
-  
+
   <build>
     <plugins>
       <plugin>
@@ -127,10 +128,22 @@
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <argLine>
-            --add-opens java.base/java.lang=ALL-UNNAMED
+            ${java11.opensargs}
           </argLine>
         </configuration>
       </plugin>
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>not-java8</id>
+      <activation>
+        <jdk>[11,]</jdk>
+      </activation>
+      <properties>
+        <java11.opensargs>--add-opens java.base/java.lang=ALL-UNNAMED</java11.opensargs>
+      </properties>
+    </profile>
+  </profiles>
 </project>

